odoo/addons/base/tests/test_db_cursor.py

```diff
@@ -115,3 +115,32 @@ def test_sub_rollback(self):
 
         self.cr.rollback()
         self.check(self.record, 'A')
+
+    def test_hooks(self):
+        log = []
+
+        def make_hook(msg):
+            def hook():
+                log.append(msg)
```

Comment:
Oh my, I forgot a `return hook` here :wink:

Reply:
And you should replace `with ... as cr` by `cr = ...`, because the closing of the `with` will try to commit the closed cursor!

Reply:
Sorry I threw some code at you without testing :frowning_face:

Reply:
I just tested it on my machine, and now it works :+1:

Reply:
And I committed without testing....